Tempo/PlayerNavigationController.swift

```diff
@@ -8,25 +8,78 @@
 
 import UIKit
 
-class PlayerNavigationController: UINavigationController {
+let PostLikedStatusChangeNotification = "PostLikedStatusChange"
+
+protocol PostDelegate {
+	var currentPost: Post? { get }
+}
+
+class PlayerNavigationController: UINavigationController, PostDelegate {
 
 	var playerCell: PlayerCellView!
 	let frameHeight = CGFloat(72)
 	
+	var expandedCell: ExpandedPlayerView!
+	let expandedHeight = CGFloat(198)
+	
+	var postsRef: [Post]?
+	var postRefIndex: Int?
+	var currentPost: Post? {
+		didSet {
+			if let newPost = currentPost {
+				playerCell.updateCellInfo(newPost)
+				expandedCell.updateCellInfo(newPost)
+			}
+		}
+	}
+	
 	override func viewDidLoad() {
 		super.viewDidLoad()
-		let playerFrame = UIView(frame: CGRectMake(0, UIScreen.mainScreen().bounds.height - 72, UIScreen.mainScreen().bounds.width, frameHeight))
+		let playerFrame = UIView(frame: CGRectMake(0, UIScreen.mainScreen().bounds.height - frameHeight, UIScreen.mainScreen().bounds.width, frameHeight))
 		playerFrame.backgroundColor = UIColor.redColor()
-		self.view.addSubview(playerFrame)
-		playerCell = NSBundle.mainBundle().loadNibNamed("PlayerCellView", owner: self, options: nil).first as! PlayerCellView
-		playerCell.setup()
-		playerCell.frame = CGRectMake(0, 0, UIScreen.mainScreen().bounds.width, frameHeight)
-		playerFrame.addSubview(playerCell)
+		view.addSubview(playerFrame)
+		playerCell = NSBundle.mainBundle().loadNibNamed("PlayerCellView", owner: self, options: nil).first as? PlayerCellView
+		playerCell?.setup(self)
+		playerCell?.frame = CGRectMake(0, 0, UIScreen.mainScreen().bounds.width, frameHeight)
+		playerCell?.userInteractionEnabled = false
+		playerFrame.addSubview(playerCell!)
 		// Do any additional setup after loading the view.
+		
+		// Setup expandedCell
+		expandedCell = NSBundle.mainBundle().loadNibNamed("ExpandedPlayerView", owner: self, options: nil).first as? ExpandedPlayerView
+		expandedCell?.setup(self)
+		expandedCell?.frame = CGRectMake(0, UIScreen.mainScreen().bounds.height, UIScreen.mainScreen().bounds.width, expandedHeight)
+		view.addSubview(expandedCell!)
+		
+		
+		NSNotificationCenter.defaultCenter().addObserverForName(PlayerDidFinishPlayingNotification, object: nil, queue: nil) { [weak self] note in
```

Comment:
If you are going to call `addObserver` and assign that to an observer instance variable, that same observer should receive a `removeObserver` call before the next assignment from another addObserver call to that same observer variable. Ideally, observers are added once on initialization and removed on cleanup. I'd figure out what the lifecycle is of any particular class it might be in, but try your best to avoid multiple places you call `addObserver`.